downloads/swan-lake-release-notes/swan-lake-2201.3.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,468 @@
+- Added support for populating the HATEOAS link's `types` field based on the resource return type
+
+#### `graphql` package
+
+- Added support for disabling introspection queries
+- Added support for GraphQL interfaces
+- Added support for interfaces implementing interfaces
+- Introduced GraphQL client configurations
+
+#### `gRPC` package
+
+- Added server reflection support for gRPC services
+
+### Improvements
+
+##### `http` package
+
+- Reduced the listener default timeout to 60s and client default timeout to 30s
+- Improved data binding based on the MIME type match
+
+#### `graphql` package
+
+- Added service-level interceptor execution for records fields, maps, and tables
+- Added service-level interceptor execution for subscriptions
+- Changed to return all the errors related to a GraphQL document in a single response
+
+#### `gRPC` package
+
+- Updated Protocol Buffers version to 3.21.7
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.3.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-standard-library/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AType%2FBug+is%3Aclosed).
+
+## Developer tools updates
+
+### New features
+
+#### Language Server
+
+- Introduced a new `Extract to local variable` code action
+- Introduced a new `Extract to constant` code action
+- Introduced a new `Extract to function` code action
+- Introduced a new code action to convert an array to array mapping into a query expression
+- Introduced the `loadProject()` API to the workspace manager to open up a project programmatically
+
+#### CLI
+
+##### Native-image build (Experimental)
+
+Introduced the `--native` flag, which generates a GraalVM native executable when building a Ballerina project.
+
+- The generated executable contains the modules in the current package, their dependencies, Ballerina runtime, and statically linked native code from the JDK
+
+- The new CLI command will be `bal build --native`
+
+>**Note:** There is a known [issue](https://github.com/ballerina-platform/ballerina-lang/issues/38665) in the native image build when using locks along with loops. 
+
+#### Debugger
+
+Introduced the `run-in-terminal` feature, which provides the capability to debug Ballerina programs that take user inputs in the launch mode. This can be achieved by setting the `terminal` attribute in the launch configurations to `integrated`. Once the configuration is set, launching the debugger will cause the Ballerina programs to be executed in a separate integrated VS Code terminal.
+
+### Improvements
+
+#### Language Server
+
+- Improved completion sorting within the expression of the `if` condition
+- Improved the `Change variable type` code action
+
+#### OpenAPI Tool
+
+Changed the default behaviour of the OpenAPI client generation to generate `resource` functions instead of `remote` functions. If you need to generate a `remote` function, use the `--client-methods=remote` flag in the OpenAPI to Ballerina CLI commands.
```

Comment:
Shall we change this to something like below? @praneesha WDYT?
```suggestion
- Improved CLI client generation to generate Ballerina client resource functions by default. (However, the users can use `--client-methods=remote` command option to switch back to client remote functions instead.)
```

Reply:
```suggestion
Improved CLI client generation to generate the Ballerina client resource methods by default. (However, the users can use `--client-methods=remote` command option to switch back to client remote functions instead.)
```